plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/action/CompareWithLatestAction.java

```diff
@@ -91,21 +87,11 @@ public void actionPerformed(ActionEvent e) {
                        dialogFactory.createMessageDialog(locale.compareMessageIdenticalContentTitle(),
                                                          locale.compareMessageIdenticalContentText(), null).show();
                    } else {
-                       final String[] changedFiles = diff.split("\n");
-                       if (changedFiles.length == 1) {
-                           project.getFile(changedFiles[0].substring(2)).then(file -> {
-                               if (file.isPresent()) {
-                                   comparePresenter.showCompareWithLatest(file.get(),
-                                                                          defineStatus(changedFiles[0].substring(0, 1)),
-                                                                          REVISION);
-                               }
-                           });
+                       ChangedItems changedItems = new ChangedItems(project, diff);
+                       if (changedItems.getFilesQuantity() == 1) {
+                           comparePresenter.showCompareWithLatest(changedItems, null, REVISION);
                        } else {
-                           Map<String, Status> items = new HashMap<>();
-                           for (String item : changedFiles) {
-                               items.put(item.substring(2, item.length()), defineStatus(item.substring(0, 1)));
-                           }
-                           changesListPresenter.show(items, REVISION, null, project);
+                           changesListPresenter.show(changedItems, REVISION, null);
```

Comment:
Good piece of advice, but not for this case, unfortunately...
Otherwise we have to make additional methods with different names (to cover two nullable parameters), which will cause additional logic on the invocations.